app-crypt/tpm2-openssl/tpm2-openssl-1.1.1.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="OpenSSL Provider for TPM2 integration"
+HOMEPAGE="https://github.com/tpm2-software/tpm2-openssl"
+SRC_URI="https://github.com/tpm2-software/tpm2-openssl/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64"
+IUSE="test"
+
+RDEPEND=">=app-crypt/tpm2-tss-3.2.0:=
+	>=dev-libs/openssl-3:0="
+
+DEPEND="${RDEPEND}
+	test? ( app-crypt/swtpm
+		app-crypt/tpm2-abrmd
+		app-crypt/tpm2-tools
+	 )
+"
+
+BDEPEND="virtual/pkgconfig"
+
+RESTRICT="!test? ( test )"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.1.1-tests-include-base-provider-required-to-load-ecparam.patch"
+)
+
+src_test() {
+	dbus_run() {
+		(
+			# start isolated dbus session bus
+			dbus_data=$(dbus-launch --sh-syntax) || exit
+			eval "${dbus_data}"
+
+			$@
+			ret=${?}
+
+			kill "${DBUS_SESSION_BUS_PID}"
+			exit "${ret}"
+		) || die
+	}
+
+	tpm2_run_with_emulator() {
+		export XDG_CONFIG_HOME="${T}/.config/swtpm"
+		"${BROOT}"/usr/share/swtpm/swtpm-create-user-config-files || die
+
+		mkdir -p "${XDG_CONFIG_HOME}/mytpm1" || die
+		swtpm_setup_args=(
+			--tpm2
+			--tpmstate "${XDG_CONFIG_HOME}/mytpm1"
+			--createek
+			--allow-signing
+			--decryption
+			--create-ek-cert
+			--create-platform-cert
+			--lock-nvram
+			--overwrite
+			--display
+		)
+		swtpm_setup "${swtpm_setup_args[@]}" || die
+
+		swtpm_socket_args=(
+			--tpm2
+			--tpmstate dir="${XDG_CONFIG_HOME}/mytpm1"
+			--flags startup-clear
+			--ctrl type=unixio,path="${XDG_CONFIG_HOME}/mytpm1/swtpm.socket.ctrl"
+			--server type=unixio,path="${XDG_CONFIG_HOME}/mytpm1/swtpm.socket"
+			--pid file="${XDG_CONFIG_HOME}/mytpm1/swtpm.pid"
+			--daemon
+		)
+		swtpm socket "${swtpm_socket_args[@]}" || die
+
+		tpm2_abrmd_args=(
+			--logger=stdout
+			--tcti=swtpm:path="${XDG_CONFIG_HOME}/mytpm1/swtpm.socket"
+			--session
+			--flush-all
+		)
+		tpm2-abrmd "${tpm2_abrmd_args[@]}" &
+
+		export TPM2OPENSSL_TCTI="tabrmd:bus_type=session"
+		export TPM2TOOLS_TCTI="tabrmd:bus_type=session"
+
+		$@ || die
+
+		# When swtpm dies, tmp2-abrmd will exit
+		kill $(< "${XDG_CONFIG_HOME}/mytpm1/swtpm.pid") || die
+	}
+
+	dbus_run tpm2_run_with_emulator make check
+}
```

Comment:
```
>>> Test phase: app-crypt/tpm2-openssl-1.1.1
Writing /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/.config/swtpm/swtpm_setup.conf.
Writing /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/.config/swtpm/swtpm-localca.conf.
Writing /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/.config/swtpm/swtpm-localca.options.
Starting vTPM manufacturing as dseifert:dseifert @ Mon 11 Sep 2023 11:42:51 AM CEST
TPM is listening on Unix socket.
Successfully created RSA 2048 EK with handle 0x81010001.
Successfully created NVRAM area 0x1c00004 for RSA 2048 EK template.
  Invoking /usr/bin/swtpm_localca --type ek --ek b7256834ce3e347a7e21459dc8ad95280505d38061b17665d8adbbeeeb6a5d1e382cb711f18feca81a54b709f5c3ca67d9754dd63850ff78b047310170e5fc7dfecfe286dc816c9e0d96ef6b911ebc00e9ec8dae68789dd35f9bf6ebc1e22899a1ff6faf5b7a96dabde2d3baf6925de68f2fa7746e374c24a71b98a586aa1a99c54b7a6477a70d7cb7d70fc16b2d1ec811b20b95d6bfdb81ddb7b4f62f6947db6eeff282bf851c5af1276bde6fa94b3d9dbb131de0e71884c904edfda54cf0c7db59b71827d011cc5f1c45f0b2c5c85d9a7b0cedddbefd384528eabaac28b4bb545119b1b2374c88cb2d9b324b4f1a9f95514c9b22fbf05593e0bf3f1bf982bb --dir /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/swtpm_setup.certs.PX0TA2 --tpm-spec-family 2.0 --tpm-spec-level 0 --tpm-spec-revision 164 --tpm-manufacturer id:00001014 --tpm-model swtpm --tpm-version id:20191023 --tpm2 --configfile /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/.config/swtpm/swtpm-localca.conf --optsfile /var/tmp/portage/app-crypt/tpm2-openssl-1.1.1/temp/.config/swtpm/swtpm-localca.options
swtpm_localca exit with status 1: Could not find swtpm_cert in PATH.
```
no idea how this could've worked in the first place

Reply:
It does work, but when gnutls USE flag was removed from app-crypt/swtpm, it broke. See https://github.com/gentoo/gentoo/pull/32704 .

Reply:
The gnutls flag has been re-added to swtpm. and I've added the appropriate dependency. It should work now.